xarray/core/indexing.py

```diff
@@ -383,11 +383,14 @@ class BasicIndexer(ExplicitIndexer):
 
     __slots__ = ()
 
-    def __init__(self, key: tuple[int | np.integer | slice, ...]):
+    def __init__(self, key: tuple[int | np.integer | slice, ...], *, _fastpath=False):
```

Comment:
nit: I _think_ we generally use `fastpath` rather than `_fastpath`, though no strong view. We could also have a different method `from_key` given there isn't much overlap

Reply:
sure, happy to change these variable names.

Reply:
yeah, i'm thinking through some of the implications here, lets review at the end of the day.